packages/webapp/src/pages/chat/Chat.tsx

```diff
@@ -59,13 +65,117 @@ const Chat = () => {
+      AccentDark: '#fdfeff',
+      TextColor: '#fdfeff',
+      BackgroundColor: '#32343e',
+      ForegroundColor: '#4e5288',
+      FormBackgroundColor: '#32343e',
+      BorderRadius: '10px',
+      BorderWidth: '3px',
+      FontBaseSize: '14px',
+    };
+    const defaultStyles = isDarkTheme ? styleDefaultsDark : styleDefaultsLight;
+    const storedStyles = localStorage.getItem('customStyles');
+    return storedStyles ? JSON.parse(storedStyles) : defaultStyles;
+  });
+
+  const handleCustomStylesChange = (newStyles: CustomStylesState) => {
+    setCustomStyles(newStyles);
+  };
+
+  const handleThemeToggle = (newIsDarkTheme: boolean) => {
+    // Get the ChatComponent instance (modify this according to how you manage your components)
+    const chatComponent = document.querySelector('chat-component');
+    if (chatComponent) {
+      // Remove existing style attributes
+      chatComponent.removeAttribute('style');
+      // eslint-disable-next-line unicorn/prefer-dom-node-dataset
+      chatComponent.setAttribute('data-theme', newIsDarkTheme ? 'dark' : '');
+    }
+    // Update the body class and html data-theme
+    localStorage.removeItem('customStyles');
+
+    // Update the state
+    setIsDarkTheme(newIsDarkTheme);
+  };
+
+  useEffect(() => {
+    // Update the state when local storage changes
+    const handleStorageChange = () => {
+      const storedStyles = localStorage.getItem('customStyles');
+      if (storedStyles) {
+        setCustomStyles(JSON.parse(storedStyles));
+      }
+
+      const storedBranding = localStorage.getItem('isBrandingEnabled');
+      if (storedBranding) {
+        setEnableBranding(JSON.parse(storedBranding));
+      }
+
+      const storedTheme = localStorage.getItem('isDarkTheme');
+      if (storedTheme) {
+        setIsDarkTheme(JSON.parse(storedTheme));
+      }
+    };
+
+    // Attach the event listener
+    window.addEventListener('storage', handleStorageChange);
+
+    // Store customStyles in local storage whenever it changes
+    localStorage.setItem('customStyles', JSON.stringify(customStyles));
+
+    // Store isBrandingEnabled in local storage whenever it changes
+    localStorage.setItem('isBrandingEnabled', JSON.stringify(isBrandingEnabled));
+
+    // Store isDarkTheme in local storage whenever it changes
+    localStorage.setItem('isDarkTheme', JSON.stringify(isDarkTheme));
+
+    // Scroll into view when isLoading changes
+    chatMessageStreamEnd.current?.scrollIntoView({ behavior: 'smooth' });
+    // Toggle 'dark' class on the shell app body element based on the isDarkTheme prop and isConfigPanelOpen
+    document.body.classList.toggle('dark', isDarkTheme);
+    document.documentElement.dataset.theme = isDarkTheme ? 'dark' : '';
+    // Clean up the event listener when the component is unmounted
+    return () => {
+      window.removeEventListener('storage', handleStorageChange);
+    };
+  }, [customStyles, isBrandingEnabled, isDarkTheme, isLoading]);
```

Comment:
do you need to replicate this in OneShot.tsx? If so, I think it might be better to move all this into it's own context. Ideally themes in React should be passed in as a context provider pattern so that root level themes can be passed in and can be updated like that across all the child components.

Reply:
Ok, I will also handle this as part of the theming refactoring, before making this PR ready for review.